duckduckgo/iOS
# 16. Subscription: Display "Activation in progress" message

Task/Issue URL: https://app.asana.com/0/414235014887631/1206362842825241/f

**Description**:
- Display an "Activation in progress" message if there's an active subscription but no available entitlements.

This situation can be caused by the Backend not receiving Apple transaction information.   While the app will continue to fetch the entitlements every time the user access "Settings", the user can also initial a manual restore from the error, which should force the transaction to be synced again.


**Steps to test this PR**:
1. Get a subscription or restore a purchase 
2. Go To line 166 in `SettingsSubscriptionView` 
3.  Replace `purchaseSubscriptionView` with `noEntitlementsAvailableView` ([line 168](https://github.com/duckduckgo/iOS/pull/2535/files#diff-475c1cc6b87611a8d8cfe20ab0160d3c1454dc278de4e4ef77f74ed749121e51R168))
4. Run the app again
5. Open Settings
6. Confirm the message (below) is displayed in settings.






---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
